java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -287,30 +348,92 @@ private API getXPAPI(ExpressionTree et) {
       if (!mit.getMethodSelect().getKind().equals(Kind.MEMBER_SELECT)) {
         return API.UNKNOWN;
       }
+      MemberSelectTree mst = (MemberSelectTree) mit.getMethodSelect();
+      String methodName = mst.getIdentifier().toString();
+      List<Map<String, Object>> methodPropertiesForMethodName =
+          configMethodProperties.get(methodName);
+      if (methodPropertiesForMethodName != null) {
+        return getAPIFromMethodPropsAndMIT(mit, methodPropertiesForMethodName);
+      }
+    }
+    return API.UNKNOWN;
+  }
 
-      if (mit.getArguments().size() == 1 || mit.getArguments().size() == 2) {
-        ExpressionTree arg = mit.getArguments().get(0);
-        Symbol argSym = ASTHelpers.getSymbol(arg);
-        if (isLiteralTreeAndMatchesFlagName(arg)
-            || isVarSymbolAndMatchesFlagName(argSym)
-            || isSymbolAndMatchesFlagName(argSym)) {
-          MemberSelectTree mst = (MemberSelectTree) mit.getMethodSelect();
-          String methodName = mst.getIdentifier().toString();
-          if (controlMethods.contains(methodName)) {
-            return API.IS_CONTROL;
-          } else if (treatedMethods.contains(methodName)) {
-            return API.IS_TREATED;
-          } else if (deleteMethods.contains(methodName)) {
-            return API.DELETE_METHOD;
-          } else if (treatmentGroupMethods.contains(methodName)) {
-            return API.IS_TREATMENT_GROUP_CHECK;
+  private API getAPIFromMethodPropsAndMIT(
+      MethodInvocationTree mit, List<Map<String, Object>> methodPropertiesForMethodName) {
+    for (Map<String, Object> currentMethodProperties : methodPropertiesForMethodName) {
+      boolean returnSpecified =
+          checkValueStringExistsAndNotEmpty(currentMethodProperties, RETURN_TYPE_STRING);
+      boolean receiverSpecified =
+          checkValueStringExistsAndNotEmpty(currentMethodProperties, RECEIVER_TYPE_STRING);
+      boolean argumentIndexSpecified = checkArgumentIndexExistsAndIsValid(currentMethodProperties);
```

Comment:
instead of `checkArgumentIndex` returning `boolean`, return the appropriate object. Then, we can avoid line 374.